galley/pkg/mcp/server/server_test.go

```diff
@@ -57,7 +57,7 @@ func (config *mockConfigWatcher) Watch(req *v2.DiscoveryRequest, out chan<- *Wat
 		rsp.TypeURL = req.TypeUrl
 		return rsp, nil
 	} else if config.closeWatch {
-		close(out)
+		go func() { out <- nil }()
```

Comment:
Why is this in a goroutine? is there concern with blocking?

Reply:
Sending an immediate response via the channel will block the caller. The contract for `Watch` says the caller should _return_ a response immediately or deliver a response later via the channel. This could be improved in a follow-up PR. The alternative is a response channel per type which, given the dynamic nature of supported types, means we'll need to use reflection to dynamic select over a set of channels (https://golang.org/pkg/reflect/#Select) or demux the channels ourselves with a goroutine per channel.